src/Battlescape/ActionMenuState.cpp

```diff
@@ -180,6 +180,44 @@ ActionMenuState::ActionMenuState(BattleAction *action, int x, int y) : _action(a
 	{
 		addItem(BA_USE, "STR_USE_MIND_PROBE", &id, Options::keyBattleActionItem1);
 	}
+	else if (weapon->getBattleType() == BT_SCRIPTED)
+	{
+		if (weapon->getCostUse().Time > 0 || weapon->getCostUse().Mana > 0)
+		{
+			addItem(BA_USE, "STR_USE_SCRIPTED", &id, Options::keyBattleActionItem1);
+		}
+		else if (weapon->getCostSnap().Time > 0 || weapon->getCostSnap().Mana > 0)
```

Comment:
why else if? I'm pretty sure "fully configurable actions" for modders means they can make a weapon with any action combination they like. With these else ifs, it looks like they are stuck with normal already existing firearms (aimed, snap, auto) or normal already existing psi amps (use, panic, MC)

Reply:
I was under the impression, that certain limitations did apply about the slots for the actions. Maybe I misunderstood that part and this could be made even more flexible. Currently it allows for example a PSI Action, a firearm action and a general use action. They are not stuck, because the script can catch each of the actions and re-define them on the fly. That was the idea at least.

Reply:
I'm just casually reading the code, didn't investigate much. But on the first look, it looks like you cannot have for example a Use action and a Snapshot action at the same time. True or false?

Reply:
False. That is exactly one of the uses cases. The actions as defined here in the menuState can be overridden by the use script. So you could define 5 actions and the script can the choose what these 5 actions are. You could have them all be snapshots or all be use actions, by return the corresponding values. You could also just keep the actions as defined and let them trough. Or you could redefine at will. Leaving the use action be and maybe changing the aimed shot to a snap shot.

Reply:
I remember now, why I did these if/elses: Options::keyBattleActionItem1. I first thought these specified the position in the menu and so I could not have two actions with the same key. I only later realized, that these are actually just keyboard hotkey bindings. It's still worth thinking about having them bound in a sensible way. I also tried limited the item to 5 actions because of this. But I agree, we could easily have this even more flexible and remove some of the limitations I introduced under a false premise.

Reply:
Adding something as an aimed shot just to "convert" it later to a snap shot by a script is just... stupid, sorry can't find a better word, no offense intended. I'd suggest to just keep adding defined actions one by one, until you reach 5 actions and then stop adding.

Reply:
There might be use cases for that scenario: SnapShot and AimedShot defined with different accuracy. SnapShot gets triggered on reaction fire, a skill triggers "50% chance to fire with aimed mode on reaction shots". The script can then throw the dice and decide wether to do a snap shot or an aimed shot.

Reply:
Yes, but it should be an exception, not a rule. The current "else if" logic would mean that everyone wanting to use a Use and Snapshot actions would need to make such a workaround with redefining actions. All I wanted to say is... get rid of the "else", add actions as long as you can (i.e. until you reach 5). Modders will understand the limit of 5 easier than the redefinition logic.

Reply:
Will do